### ISSUE TITLE
Non-interactive install for `kup`

### ISSUE THREAD

**Issue by ehildenb** [runtimeverification/kup#56]
I'd like a fully non-interactive option for installing `kup` that can be used in teh docker image creation setting: https://github.com/runtimeverification/evm-semantics/pull/1747

In particular, if ew could do `bash <(curl ...) --non-interactive` to install `kup`, that would be very helpful.

In that linked PR, you can clone KEVM, and in the root of th edirectory make a script called `bar` which contains the docker image ceration code:

```
          set -euxo pipefail
          version=$(cat package/version)
          tag_name="kevm:${version}"
          container_name="kevm-ci-test-${GITHUB_SHA}"
          docker build . --tag ${tag_name} --file package/docker/Dockerfile
          docker run                        \
            --name ${container_name}        \
            --rm                            \
            --interactive                   \
            --tty                           \
            --detach                        \
            --user root                     \
            --workdir /home/user            \
            ${tag_name}
          docker exec -u user ${container_name} /bin/bash -c 'bash <(curl https://kframework.org/install)'
          docker exec -u user ${container_name} /bin/bash -c "kup install kevm --version v${version}"
```

Basically, that script should work to make a docker image taht then has KEVM installed.

**Comment by goodlyrottenapple** [comment]
Given various constraints of docker images, it might b best to forego using our install script and just install nix followed by installing kup.

For the first step, refer to https://github.com/DeterminateSystems/nix-installer. They have an example for docker, which i've adapted for our potential use of kup:

```docker
# Dockerfile
FROM ubuntu:latest
RUN apt update -y
RUN apt install curl -y
RUN curl --proto '=https' --tlsv1.2 -sSf -L https://install.determinate.systems/nix | sh -s -- install linux --init none --no-confirm
ENV PATH="${PATH}:/nix/var/nix/profiles/default/bin"
RUN nix shell github:runtimeverification/kup#kup  --option extra-substituters 'https://k-framework.cachix.org' --option extra-trusted-public-keys 'k-framework.cachix.org-1:jeyMXB2h28gpNRjuVkehg+zLj62ma1RnyyopA/20yFE='
...kup should be on the path form here...
```

**Comment by goodlyrottenapple** [comment]
@ehildenb can this be closed?
